src/main/java/sparksoniq/jsoniq/runtime/iterator/primary/ObjectConstructorRuntimeIterator.java

```diff
@@ -76,25 +78,23 @@ public ObjectItem next() {
                 for (RuntimeIterator valueIterator : this._values) {
                     List<Item> currentResults = new ArrayList<>();
                     valueIterator.open(this._currentDynamicContext);
-                    while (valueIterator.hasNext())
+                    while (valueIterator.hasNext()) {
                         currentResults.add(valueIterator.next());
-//                    if(valueIterator.hasNext())
-//                        throw new IteratorFlowException("Object value must return one item!");
+                    }
                     valueIterator.close();
                     //SIMILAR TO ZORBA, if value is more than one item, wrap it in an array
-                    if (currentResults.size() > 1)
+                    if (currentResults.size() > 1) {
                         values.add(new ArrayItem(currentResults));
-                    else
+                    } else if (currentResults.size() == 1) {
                         values.add(currentResults.get(0));
+                    } else {
+                        values.add(new NullItem());
+                    }
```

Comment:
Object construction would cause exceptions for empty sequences used as values. I have added a size check and put in NullItem for empty sequences. I believe that this behavior is compliant with JSONiq specifications at:
http://www.jsoniq.org/docs/JSONiq/html-single/index.html#idm71239520

> And do not worry about the value expression: if it is is empty, null will be used as a value, and if it contains two items or more, they will be wrapped into an array.